src/components/login/login.spec.js

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import { Provider } from 'react-redux';
+import thunk from 'redux-thunk';
+import { BrowserRouter as Router } from 'react-router-dom';
+import configureStore from 'redux-mock-store';
+import LoginForm, { LoginForm as UnconnectedLoginForm } from './login';
+
+const authReducer = {
+  isAuthenticated: false,
+  user: {},
+  status: 'rest'
+};
+
+const middlewares = [thunk];
+const mockStore = configureStore(middlewares);
+const store = mockStore({
+  authReducer
+});
+
+const props = {
+  LoginForm: jest.fn()
+};
+
+const unconnectedProps = {
+  authAction: jest.fn(),
+  status: '',
+  isAuthenticated: false,
+  error: null,
+  location: { url: 'http://localhost' }
+};
+
+const setUp = () =>
+  mount(
+    <Provider store={store} {...props}>
+      <Router>
+        <LoginForm />
+      </Router>
+    </Provider>
+  );
+
+describe('Login Component', () => {
+  it('should render without crashing', () => {
+    const wrapper = setUp();
+    expect(wrapper).toBeDefined();
+    expect(wrapper.find('LoginForm')).toBeTruthy();
+    expect(wrapper.find('div')).toBeTruthy();
+  });
+
+  it('should render one form', () => {
+    const wrapper = setUp();
+    expect(wrapper.find('form').length).toBe(1);
+  });
+
+  it('should render 3 input fields', () => {
+    const wrapper = setUp();
+    expect(wrapper.find('input').length).toBe(3);
+    expect(
+      wrapper
+        .find('input')
+        .at(0)
+        .props().type
+    ).toBe('email');
+    expect(
+      wrapper
+        .find('input')
+        .at(1)
+        .props().type
+    ).toBe('password');
+    expect(
+      wrapper
+        .find('input')
+        .at(2)
+        .props().type
+    ).toBe('submit');
+  });
+
+  it('should redirect to Profile if user is logged in already', () => {
+    authReducer.isAuthenticated = true;
+    const wrapper = setUp();
+    expect(wrapper).toBeDefined();
+    authReducer.isAuthenticated = true;
```

Comment:
Mutating global variables is a bad idea. It can lead to inconsistent results, especially in a multi-threaded or parallel programming environment.

See @Onyimatics for the correct way to go about this